account_invoice_pricelist/models/account_invoice_line.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, api
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.onchange('product_id')
```

Comment:
You have to recompute also on quantity change, as there are pricelist with escalated prices.

Reply:
We could maybe make depends on invoice_id.pricelist_id too. It make the "update from pricelist" on the invoice obsolete.
* I think it more usable. I think we want to apply the allways pricelist when we change the pricelist
* it's not the same behaviour like in sale module. (odoo core).